src/node_api_backport.cc

```diff
@@ -0,0 +1,146 @@
+#include "node_api_backport.h"
+
+using v8::Function;
+using v8::HandleScope;
+using v8::Local;
+using v8::MaybeLocal;
+using v8::Object;
+using v8::TryCatch;
+using v8::Value;
+
+namespace node {
+
+CallbackScope::CallbackScope(v8::Isolate *_isolate,
+                             v8::Local<v8::Object> _object,
+                             node::async_context context) :
+    isolate(_isolate),
+    env(node::Environment::GetCurrent(isolate->GetCurrentContext())),
+    _try_catch(isolate),
+    object(_object),
+    callback_scope(
+        node::Environment::GetCurrent(isolate->GetCurrentContext())) {
+  Local<Function> pre_fn = env->async_hooks_pre_function();
+
+  Environment::AsyncCallbackScope callback_scope(env);
```

Comment:
Should this be here?  It looks like a duplicate of the callback_scope declared in the class.